### ISSUE TITLE
Broken in leaflet 1.8.0

### ISSUE THREAD

**Issue by jerome-lemaire** [smeijer/leaflet-geosearch#323]
Hi !

The plugin does not work anymore with Leaflet 1.8.0 but ok with Lealfet 1.7.1

minmal code : https://codesandbox.io/s/frosty-tamas-yevb21?file=/src/index.js

`Uncaught TypeError: Cannot read properties of undefined (reading 'indexOf')
    at NewClass.addTo (leaflet-src.js:4789:1)`

**Comment by unzsnu** [comment]
I fixed this error setting the position

```js
 const searchControl = new GeoSearchControl({
      position: 'topright', /* <---- add this, options: 'topleft', 'topright', 'bottomleft', 'bottomright' */
      provider: props.provider,
      style: props.controlStyle,
      ...props
    });
```

Position has not been set yet, so getPosition() returns undefined and then, checking the position, it breaks

**Comment by thecodemonk** [comment]
Using the sample site posted above as a test, the search doesn't actually seem to be working. I am noticing the same thing and we use MapBoxProvider. Downgrading to 1.7.1 fixed it.

**Comment by ewlarson** [comment]
I'm also seeing that search doesn't work with Leaflet v1.8.0. @unzsnu's suggestion does allow the control to render, but search isn't operable.

**Comment by brbrbr** [comment]
Appears that the default options are not initialized correctly.

I added the lot (extending the fix from [unzsnu](https://github.com/unzsnu)


```
  var searchControl = new GeoSearch.GeoSearchControl({
          provider: provider,
          position: 'topleft',
          showMarker: true,
          marker: {
            icon: get_icon('roos'),
            cat: 'roos',
            draggable: true,
          },
          maxMarker: 1,
          autoClose: true,
          autoComplete: true,
          retainZoomLevel: true,
          maxSuggestions: 5,
          keepResult: true,
          resultFormat: function(t) {
            return "" + t.result.label;
          },
          updateMap: !0

        });
```

**Comment by gregorbg** [comment]
I tried to look into this, and it seems that the `extend` call at https://github.com/smeijer/leaflet-geosearch/blob/develop/src/SearchControl.ts#L472 was changed in `leaflet` as part of the 1.8.0 release: https://github.com/Leaflet/Leaflet/commit/7b15104f2fc5e16fb23d5e13468c5df558cbf006.

Now, the extended class does not have the default options anymore (which is why the original `indexOf` error occurs), nor does it overwrite the `showResult` function properly (which is why search isn't working even if you add certain option props)

Maybe someone who is more knowledgeable than me in JS inheritance and prototypes can use this information to figure out a proper solution :)

**Comment by brbrbr** [comment]
I have no clue what all the typescript and prototyping stuff does. However, replacing the ... operators in the SearchControl.initialize function
below '_merge given options with control defaults_' with the native L.setOptions resolves the issues for me.

L.setOptions(this,options );

In the generated code it looks like:

`@@ -287,8 +287,9 @@
                                        throw new Error(x);
                                if (!t.provider)
                                        throw new Error("Provider is missing from options");
-                               this.options = r({}, this.options, {}, t),
-                               this.classNames = r({}, this.classNames, {}, t.classNames),
+
+                               this.options = e.setOptions(this, t),
+                               this.classNames = this.options.classNames, 
`

**Comment by smeijer** [comment]
Should be fixed in 3.6.1. Thanks to @vkunz (#327)

**Comment by brbrbr** [comment]
Fixes the issue. 

In the mentioned PR on leaflet, a commenter depicts that initialize should call setOptions:

https://github.com/Leaflet/Leaflet/pull/7459#pullrequestreview-586639979
